198.HouseRobber/step2.cpp

```diff
@@ -0,0 +1,21 @@
+class Solution {
+public:
+  int rob(vector<int>& nums) {
+    if (nums.size() == 1) {
+      return nums[0];
+    }
+
+    // loopの中での2つ前の状態
+    int second_previous_max_money = nums[0];
+    // loopの中での1つ前の状態
+    int previous_max_money = max(nums[0], nums[1]);
+
+    for (int i = 2; i < nums.size(); i++) {
+      int temp_money = previous_max_money;
+      previous_max_money = max(previous_max_money, second_previous_max_money + nums[i]);
+      second_previous_max_money = temp_money;
```

Comment:
個人的には、以下のほうが泥棒の作業のイメージに近いかなと思いました。が、どちらでも良いと思います
（泥棒が家`i`の前にいる時、得られる最大利得を考えていくイメージ）

```suggestion
      int curernt_max = max(previous_max_money, second_previous_max_money + nums[I]);
      second_previous_max_money = previous_max_money;
      previous_max_money = curernt_max;
```

Reply:
@Yoshiki-Iwasa 
レビューありがとうございます。こちらの方がしっくりきました。
変数宣言時は2つ前、１つ前と定義していてこの順に塗る塗らないの判断をしたかったです。